tests/test_shipment.py

```diff
@@ -1,228 +1,157 @@
-        "name": "Dr. Steve Brule",
-        "street1": "179 N Harbor Dr",
-        "city": "Redondo Beach",
-        "state": "CA",
-        "zip": "90277",
-        "country": "US",
-        "phone": "4153334444",
-        "email": "dr_steve_brule@gmail.com",
-    }
-    from_address = {
-        "name": "EasyPost",
-        "street1": "417 Montgomery Street",
-        "street2": "5th Floor",
-        "city": "San Francisco",
-        "state": "CA",
-        "zip": "94104",
-        "country": "US",
-        "phone": "4153334444",
-        "email": "support@easypost.com",
-    }
-    parcel = {
-        "length": 20.2,
-        "width": 10.9,
-        "height": 5,
-        "weight": 65.9,
-    }
+def test_shipment_smartrate(basic_shipment):
+    shipment = easypost.Shipment.create(**basic_shipment)
 
-    shipment = easypost.Shipment.create(
-        to_address=to_address,
-        from_address=from_address,
-        parcel=parcel,
-    )
-    assert shipment.rates
+    assert shipment.rates is not None
 
     smartrates = shipment.get_smartrates()
-    assert shipment.rates[0]["id"] == smartrates[0]["id"]
+
     assert smartrates[0]["time_in_transit"]["percentile_50"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_75"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_85"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_90"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_95"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_97"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_99"] is not None
+
+
+@pytest.mark.vcr()
+def test_shipment_create_empty_objects(basic_shipment):
+    shipment_data = basic_shipment
+    shipment_data["options"] = None
+    shipment_data["tax_identifiers"] = None
+    shipment_data["reference"] = ""
+
+    shipment = easypost.Shipment.create(**shipment_data)
+
+    assert isinstance(shipment, easypost.Shipment)
+    assert str.startswith(shipment.id, "shp_")
+    assert shipment.options is not None
+    assert shipment.customs_info is None
+    assert shipment.reference is None
+
+
+@pytest.mark.vcr()
+def test_shipment_create_tax_identifier(basic_shipment, tax_identifier):
+    shipment_data = basic_shipment
+    shipment_data["tax_identifiers"] = [tax_identifier]
+
+    shipment = easypost.Shipment.create(**shipment_data)
+
+    assert isinstance(shipment, easypost.Shipment)
+    assert str.startswith(shipment.id, "shp_")
+    assert shipment.tax_identifiers[0]["tax_id_type"] == "IOSS"
+
+
+@pytest.mark.vcr()
+def test_shipment_create_with_ids(basic_address, basic_parcel):
+    from_address = easypost.Address.create(**basic_address)
+    to_address = easypost.Address.create(**basic_address)
+    parcel = easypost.Parcel.create(**basic_parcel)
+
+    shipment = easypost.Shipment.create(
+        from_address={"id": from_address.id}, to_address={"id": to_address.id}, parcel={"id": parcel.id}
```

Comment:
Please always use trailing commas, Black will format this accordingly when you do so (breaking these up to their own lines for better readability).

```suggestion
        from_address={"id": from_address.id}, to_address={"id": to_address.id}, parcel={"id": parcel.id},
```